supercommand_test.go

```diff
@@ -192,35 +200,31 @@ func (s *SuperCommandSuite) TestVersionFlag(c *gc.C) {
 	testVersionFlagCommand := &testVersionFlagCommand{}
 	jc.Register(testVersionFlagCommand)
 
-	var stdout, stderr bytes.Buffer
-	ctx := &cmd.Context{
-		Stdout: &stdout,
-		Stderr: &stderr,
-	}
-
 	// baseline: juju version
-	code := cmd.Main(jc, ctx, []string{"version"})
+	code := cmd.Main(jc, s.ctx, []string{"version"})
 	c.Check(code, gc.Equals, 0)
-	baselineStderr := stderr.String()
-	baselineStdout := stdout.String()
-	stderr.Reset()
-	stdout.Reset()
+	baselineStderr := cmdtesting.Stderr(s.ctx)
+	baselineStdout := cmdtesting.Stdout(s.ctx)
+
+	s.TearDownTest(c)
+	s.SetUpTest(c)
```

Comment:
Split this test up, so we don't need to do this. `TestVersionWithArg` and `TestVersionWithFlag`.